blog/2020-02-25-debezium-on-rds.adoc

```diff
@@ -0,0 +1,207 @@
+
+Periodically Debezium writes the last processed LSN and transaction id to the Kafka Connect offsets topic and advances the replication slot to match that. On startup, Debezium uses the last record from the Kafka Connect offsets topic to rewind the replication slot to the position as described before continuing streaming changes. This means that to change the position in the WAL where Debezium picks up from requires a change in both Debezium's tracked information in the Kafka Connect offsets topic as well as server side in PostgreSQL.
+
+=== Skipping Unparseable Events
+
+We were able to use the above information to make Debezium skip the unparseable event by performing the following steps:
+
+. Stop Debezium to make the replication slot inactive.
+. Check Debezium has stopped listening on the replication slot by running `SELECT * FROM pg_replication_slots WHERE slot_name = '<your-slot-name>';`. The `active` column should be `f`.
+. Check the last message in Debezium's offsets topic and note down the value for the `lsn` key. eg. `1516427642656`.
+. Convert that long representation of LSN into the hexadecimal format using PosgtreSQL's Java driver using the below Java code:
++
+[source,java]
+----
+import org.postgresql.replication.LogSequenceNumber;
+
+class Scratch {
+  public static void main(String[] args) {
+      LogSequenceNumber a = LogSequenceNumber.valueOf(1516427642656L);
+      System.out.println(a.asString());
+  }
+}
+----
+
+. Peek changes from the WAL upto the LSN above using `SELECT pg_logical_slot_peek_changes('<your-slot-name>', '<lsn-from-above>', 1)`. This is the replication change that we are going to skip, so please make sure that this is the record that you want to skip. Once confirmed, proceed to next step.
+. Advance the replication slot by skipping 1 change using `SELECT pg_logical_slot_get_changes('<your-slot-name>', NULL, 1)`. This will consume 1 change from the replication slot.
+. Publish a message to Debezium's offset topic with the next LSN and TxId. We were able to successfully get it working by adding 1 to both the `lsn` and the `txId`.
+. Deploy Debezium again and it should have skipped the record.
+
+[NOTE]
+====
+*The exact issue we hit is already fixed in Debezium 1.0 and you should update as soon as possible*.
+
+Please make sure to test the steps locally/non-production environments before performing them in production. Your mileage may vary.
+
+Also, if possible join the discussion at https://issues.redhat.com/browse/DBZ-1760[DBZ-1760] to help design a solution to the general issue of skipping unparseable records.
```

Comment:
Just merged the PR, i.e. that's fixed in the next 1.1 preview release. Could you update this accordingly? If not, no problem, I can do it, too. Thanks!

Reply:
Thanks for the quick action. I'll add a link to the PR too here.